plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,334 @@
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+        assertQueryFails("SELECT * FROM " + tableName, ".*Table '(.*)' does not exist.*");
+
+        assertQuerySucceeds(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, tableName, tableLocation));
+        String showCreateTableNew = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + " (a, b, c) COMMENT 'my-table-comment' AS VALUES (1, 'INDIA', true)");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertQuerySucceeds("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertThat(getTableComment(tableName)).isEqualTo("my-table-comment");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithDifferentTableName()
+    {
+        String tableName = "test_register_table_with_different_table_name_old_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + " AS SELECT 1 as a, 'INDIA' as b, true as c");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String showCreateTableOld = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the table content to register a new table
+        dropTableFromMetastore(tableName);
+
+        String tableNameNew = "test_register_table_with_different_table_name_new_" + randomNameSuffix();
+        assertQuerySucceeds(format("CALL %s.system.register_table('%s', '%s', '%s')", CATALOG_NAME, SCHEMA, tableNameNew, tableLocation));
+        String showCreateTableNew = (String) computeScalar("SHOW CREATE TABLE " + tableNameNew);
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew.replaceFirst(tableNameNew, tableName));
+        assertQuery("SELECT * FROM " + tableNameNew, "VALUES (1, 'INDIA', true)");
+
+        assertUpdate(format("DROP TABLE %s", tableNameNew));
+    }
+
+    @Test
+    public void testRegisterTableWithNoData()
```

Comment:
I've seen same/similar test cases in `plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java`. why is that?

Reply:
`BaseDeltaLakeRegisterTableProcedureTest` is the superset of `BaseDeltaLakeConnectorSmokeTest` for `register_table` test cases.

`BaseDeltaLakeRegisterTableProcedureTest` uses a Local File system with Flat and Glue metastore. It executes really fast and saves time while debugging. 

On another hand, `BaseDeltaLakeConnectorSmokeTest` uses Minio which takes some time to get ready for the test. But It covers cloud storage (s3, gcp, adls).

Reply:
I put only `testRegisterTable` in `BaseDeltaLakeConnectorSmokeTest`. If we want to add more tests to the smoke test, I expect it's specific to the storage rather than copy-and-paste from `BaseDeltaLakeRegisterTableProcedureTest`.